subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/CollectTeamcityMetricsWorkerAction.kt

```diff
@@ -0,0 +1,36 @@
+package com.avito.android.plugin.build_metrics
+
+import com.avito.android.plugin.build_metrics.CollectTeamcityMetricsWorkerAction.Parameters
+import com.avito.android.stats.StatsDConfig
+import com.avito.android.stats.StatsDSender
+import com.avito.teamcity.TeamcityApi
+import com.avito.teamcity.TeamcityCredentials
+import com.avito.utils.logging.CILogger
+import org.gradle.api.provider.Property
+import org.gradle.workers.WorkAction
+import org.gradle.workers.WorkParameters
+
+@Suppress("UnstableApiUsage")
+abstract class CollectTeamcityMetricsWorkerAction : WorkAction<Parameters> {
+
+    interface Parameters : WorkParameters {
+        fun getBuildId(): Property<String>
+        fun getTeamcityCredentials(): Property<TeamcityCredentials>
+        fun getStatsdConfig(): Property<StatsDConfig>
+        fun getLogger(): Property<CILogger>
+    }
+
+    override fun execute() {
```

Comment:
i really don't understand this layers logic?
- we have a task that only run worker; (but also validate some params, why?)
- we have a worker what should only run Action, but it has some non-trivial logic
- we have action that consist main logic

right?

why not put all logic to Action as it is testable without gradle?

Reply:
see docs/content/docs/ci/CustomGradlePlugins.md